### ISSUE TITLE
test_70_half_closed failing in system_tests_tcp_adaptor

### ISSUE THREAD

**Issue by ganeshmurthy** [skupperproject/skupper-router#391]
```
 =================================== FAILURES ===================================
________________________ TcpAdaptor.test_70_half_closed ________________________

self = <tests.system_tests_tcp_adaptor.TcpAdaptor testMethod=test_70_half_closed>

    def test_70_half_closed(self):
        if DISABLE_SELECTOR_TESTS:
            self.skipTest(DISABLE_SELECTOR_REASON)
        if not ncat_available():
            self.skipTest("Ncat utility is not available")
        name = "test_70_half_closed"
        self.logger.log("TCP_TEST Start %s" % name)
        self.ncat_runner(name, "INTA", "INTA", self.logger)
        self.ncat_runner(name, "INTA", "INTB", self.logger)
>       self.ncat_runner(name, "INTA", "INTC", self.logger)

/home/runner/work/skupper-router/skupper-router/skupper-router/tests/system_tests_tcp_adaptor.py:947: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <tests.system_tests_tcp_adaptor.TcpAdaptor testMethod=test_70_half_closed>
tname = 'test_70_half_closed', client = 'INTA', server = 'INTC'
logger = <system_test.Logger object at 0x7fd8632e5e40>

    def ncat_runner(self, tname, client, server, logger):
        name = "%s_%s_%s" % (tname, client, server)
        logger.log(name + " Start")
        out = self.run_ncat(TcpAdaptor.tcp_client_listener_ports[client][server], logger, data=b'abcd')
        logger.log("run_ncat returns: '%s'" % out)
>       assert len(out) > 0
E       AssertionError: assert 0 > 0
E        +  where 0 = len('')

/home/runner/work/skupper-router/skupper-router/skupper-router/tests/system_tests_tcp_adaptor.py:933: AssertionError
```
https://github.com/skupperproject/skupper-router/runs/6215745287?check_suite_focus=true#step:26:3048

This error started happening after ncat was added as a test dependency in this commit - https://github.com/skupperproject/skupper-router/commit/018170dc7ce7a71bac2b0d6313f1be7c305188e5

**Comment by jiridanek** [comment]
https://github.com/skupperproject/skupper-router/runs/6197016878?check_suite_focus=true#step:27:93

```
59: E           skrouterd: /home/runner/work/skupper-router/skupper-router/skupper-router/src/message.c:1089: qd_message_free: Assertion `DEQ_IS_EMPTY(msg->stream_data_list)' failed.
```

**Comment by kgiusti** [comment]
And the chickens have come home to roost....

Looks like the fix to #263 has caused this issue.

**Comment by jiridanek** [comment]
This issue is a very likely a duplicate of #97. That issue lacked reproduction steps, while this one is manifesting as a frequently occuring flake, so I propose to use this issue and close the old one.
